backend/src/apiserver/resource/resource_manager.go

```diff
@@ -735,9 +735,16 @@ func (r *ResourceManager) CreateJob(ctx context.Context, apiJob *api.Job) (*mode
 	if err != nil {
 		return nil, util.Wrap(err, "Create job failed")
 	}
+	var pipelineID *string
+	if apiJob.PipelineSpec != nil {
+		if apiJob.PipelineSpec.GetPipelineId() != "" {
+			pipelineID = &apiJob.PipelineSpec.PipelineId
```

Comment:
nit: I recall that we can also get pipeline id from resource ref, can you confirm?

Reply:
I dont think we can do that: 

https://github.com/kubeflow/pipelines/blob/d03277fe4ed79f2bc4780ca1cdb6bab9469eeb61/backend/src/apiserver/resource/resource_manager_util.go#L248-L250

Reply:
Thanks for confirming! You are right